### ISSUE TITLE
Exception: NoSuchMethodError com.google.android.gms.iid.MessengerCompat

### ISSUE THREAD

**Issue by marcoRS** [intercom/intercom-android#362]
## Version info
  - Intercom for Android version: 3.0.5 and 3.2.5
  - Android OS version: Multiple versions 4,5,6,7
  - Android Device: Samsung, LGE, ZTE

## Expected behavior
The app should not crash randomly citing intercom. 

## Actual behavior
The app randomly crashes when it can't find a method related to gms. 

## Steps to reproduce
 Unable to reproduce 

## Logs

```
Fatal Exception: java.lang.NoSuchMethodError: No direct method <init>(Landroid/os/Handler;)V in class Lcom/google/android/gms/iid/MessengerCompat; or its super classes (declaration of 'com.google.android.gms.iid.MessengerCompat' appears in /data/app/com.wheniwork.wiw-1/base.apk)
       at com.google.android.gms.iid.InstanceIDListenerService.<init>(Unknown Source)
       at io.intercom.android.sdk.gcm.IntercomGcmRefreshService.<init>(IntercomGcmRefreshService.java:7)
       at java.lang.Class.newInstance(Class.java)
       at android.app.ActivityThread.handleCreateService(ActivityThread.java:2859)
       at android.app.ActivityThread.-wrap4(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1427)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:148)
       at android.app.ActivityThread.main(ActivityThread.java:5417)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

**Comment by DaleCantwell** [comment]
Hi @marcoRS 
Could you let us know what versions you are using for:
`compileSdkVersion`
`targetSdkVersion`
`buildToolsVersion`
`minSdkVersion`
`'com.google.gms:google-services:3.0.0`
`com.google.firebase:firebase-messaging` or `'com.google.android.gms:play-services-gcm`
`com.android.tools.build:gradle`
This is very likely an issue with conflicting dependancy versions.

**Comment by marcoRS** [comment]
```
androidPluginVersion=2.3.3
buildToolsVersion=25.0.2
compileSdkVersion=25
minSdkVersion=19
playServicesPluginVersion=3.1.0
targetSdkVersion=25
playServicesVersion=11.0.0
```

We have the following play services and intercom dependencies:
```
compile "com.google.android.gms:play-services-auth:${playServicesVersion}"
compile "com.google.android.gms:play-services-base:${playServicesVersion}"
compile "com.google.android.gms:play-services-location:${playServicesVersion}"
compile "com.google.android.gms:play-services-places:${playServicesVersion}"
compile "com.google.android.gms:play-services-gcm:${playServicesVersion}"
compile "com.google.firebase:firebase-invites:${playServicesVersion}"
compile "com.google.firebase:firebase-messaging:${playServicesVersion}"
compile('io.intercom.android:intercom-sdk:3.2.5') { transitive = true }
```

**Comment by ronocod** [comment]
Can you try with `playServicesVersion=10.2.1`? We haven't updated to version 11 yet.

**Comment by marcoRS** [comment]
Hello, we were also observing this crash with the following version configurations: 

```
androidPluginVersion=2.2.3
buildToolsVersion=25.0.2
compileSdkVersion=25
minSdkVersion=19
playServicesPluginVersion=3.0.0
targetSdkVersion=23
playServicesVersion=10.2.4
intercomVersion=3.0.5
```

```
androidPluginVersion=2.2.3
buildToolsVersion=25.0.2
compileSdkVersion=25
minSdkVersion=19
playServicesPluginVersion=3.0.0
targetSdkVersion=23
playServicesVersion=9.4.0
intercomVersion=3.0.5
```

Unfortunately this is a crash that we can't consistently reproduce but is being experienced by our users. We upgraded intercom + google play services hoping this would fix the crash. We thought at first maybe it was a proguard issue but this is appearing in our unoptimized debug builds as well.

**Comment by ronocod** [comment]
According to [this SO answer](https://stackoverflow.com/a/38042258) this can be related to using different versions across Play Services libraries. Can you try using `playServicesVersion=10.2.1` with `intercomVersion=3.2.6`? That should make all the Play Services libraries use the same version.

**Comment by marcoRS** [comment]
We unfortunately cannot downgrade to 10.2.1 due to some compatibility issues with other libraries. How does your sdk handle the `transitive=true` , we have your sdk listed as `compile('io.intercom.android:intercom-sdk:3.2.5') { transitive = true }` If this is removed does that mean play services 10.2.1 will be used? Any plans on upgrading to 11.0.0 anytime soon?

**Comment by ronocod** [comment]
The `{ transitive = true }` no longer has any effect there, you can just use 
`compile('io.intercom.android:intercom-sdk:3.2.5')`.

Your app will only use one version of Play Services at a time, so you can't make Intercom use 10.2.1 while other libraries use 11.0.0.

We'll upgrade to 11.0.0 soon and post an update here when we do. We don't update to the newest available versions of Play Services as soon as they are released as we would create conflicts like this :(

**Comment by DaleCantwell** [comment]
Closed in [4.0.0](https://github.com/intercom/intercom-android/releases/tag/4.0.0)
